File: python/p0125.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        if not s:
            return False  
            
        if len(s) == 1:
            return True  

        l = 0  # Left pointer
        r = len(s) - 1  # Right pointer

        while l < r:
            if not s[l].isalnum():  # Skip non-alphanumeric characters on the left
                l += 1
                continue

            if not s[r].isalnum():  # Skip non-alphanumeric characters on the right
                r -= 1
                continue

            if s[l].lower() != s[r].lower():  # Check if characters are not equal ignoring case
                return False

            l += 1  
            r -= 1  

        return True 

File: javascript/p0125.js
/**
 * @param {string} s
 * @return {boolean}
 */
export var isPalindrome = function (s) {
  const regex = /^[a-zA-Z0-9]$/; // Regular expression to match alphanumeric characters

  if (!s) return false;
  if (s.length === 1) return true;

  let l = 0; // Left pointer
  let r = s.length - 1; // Right pointer

  while (l < r) {
    if (!regex.test(s[l])) {
      // Skip non-alphanumeric characters on the left
      l += 1;
      continue;
    }

    if (!regex.test(s[r])) {
      // Skip non-alphanumeric characters on the right
      r -= 1;
      continue;
    }

    if (s[l].toLocaleLowerCase() != s[r].toLocaleLowerCase()) return false; // Check if characters are not equal ignoring case

    l += 1;
    r -= 1;
  }

  return true;
};
